plugins/package-managers/gradle-inspector/src/main/kotlin/GradleInspector.kt

```diff
@@ -0,0 +1,250 @@
+import org.ossreviewtoolkit.analyzer.PackageManager
+import org.ossreviewtoolkit.downloader.VersionControlSystem
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.Issue
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageLinkage
+import org.ossreviewtoolkit.model.PackageReference
+import org.ossreviewtoolkit.model.Project
+import org.ossreviewtoolkit.model.ProjectAnalyzerResult
+import org.ossreviewtoolkit.model.Scope
+import org.ossreviewtoolkit.model.Severity
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
+import org.ossreviewtoolkit.model.config.PackageManagerConfiguration
+import org.ossreviewtoolkit.model.config.RepositoryConfiguration
+import org.ossreviewtoolkit.model.createAndLogIssue
+import org.ossreviewtoolkit.utils.ort.createOrtTempFile
+
+/**
+ * The names of Gradle (Groovy, Kotlin script) build files for a Gradle project.
+ */
+private val GRADLE_BUILD_FILES = listOf("build.gradle", "build.gradle.kts")
+
+/**
+ * The names of Gradle (Groovy, Kotlin script) settings files for a Gradle build.
+ */
+private val GRADLE_SETTINGS_FILES = listOf("settings.gradle", "settings.gradle.kts")
+
+/**
+ * The name of the option to specify the Gradle version.
+ */
+const val OPTION_GRADLE_VERSION = "gradleVersion"
+
+/**
+ * The [Gradle](https://gradle.org/) package manager for Java.
+ *
+ * This package manager supports the following [options][PackageManagerConfiguration.options]:
+ * - *gradleVersion*: The version of Gradle to use when analyzing projects. Defaults to the version defined in the
+ *   Gradle wrapper properties.
+ */
+class GradleInspector(
+    name: String,
+    analysisRoot: File,
+    analyzerConfig: AnalyzerConfiguration,
+    repoConfig: RepositoryConfiguration
+) : PackageManager(name, analysisRoot, analyzerConfig, repoConfig) {
+    private companion object : Logging
+
+    class Factory : AbstractPackageManagerFactory<GradleInspector>("GradleInspector", isEnabledByDefault = false) {
+        // Gradle prefers Groovy ".gradle" files over Kotlin ".gradle.kts" files, but "build" files have to come before
+        // "settings" files as we should consider "settings" files only if the same directory does not also contain a
+        // "build" file.
+        override val globsForDefinitionFiles = GRADLE_BUILD_FILES + GRADLE_SETTINGS_FILES
+
+        override fun create(
+            analysisRoot: File,
+            analyzerConfig: AnalyzerConfiguration,
+            repoConfig: RepositoryConfiguration
+        ) = GradleInspector(type, analysisRoot, analyzerConfig, repoConfig)
+    }
+
+    private fun extractInitScript(): File {
+        fun extractResource(name: String, target: File): File {
+            target.outputStream().use { outputStream ->
+                val resource = checkNotNull(javaClass.getResource(name)) {
+                    "Resource '$name' not found."
```

Comment:
Is this related to the IntelliJ issue you describe in the "Building" section of the README? If so, please add a hint here that this can be fixed by running those Gradle tasks.

Reply:
This was rather something I added during development, and to deal with a non-nullable resource value. In any case, I think it's probably not a good idea to put developer-hints into this user-facing message.